Src/LogData.c

```diff
@@ -230,29 +234,55 @@ void logDataTask(void const* arg)
 
     for (;;)
     {
-        switch (getCurrentFlightPhase())
-        {
-            case PRELAUNCH:
-                lowFrequencyLogToSdRoutine(data, buffer, PRELAUNCH);
-                break;
-
-            case BURN:
-            case COAST:
-            case DROGUE_DESCENT:
-                highFrequencyLogToSdRoutine(data, buffer);
-                break;
-
-            case MAIN_DESCENT:
-                lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT);
-                break;
-
-            case ABORT:
-                lowFrequencyLogToSdRoutine(data, buffer, ABORT);
-                break;
-
-            default:
-                lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT); // won't work
-                break;
+        checkForHeader();
```

Comment:
I think it is unnecessary to checkForHeader every time we go through this loop. I think this one can be removed, and then we can just check for it in the ```else if(containsHeader ==  0)```